File: src/StealME.Server/StealME.Server.Messaging/TypeResolver.cs
using System;
using System.Collections.Generic;
using System.Linq;
using StealME.Networking.Protocol;
using StealME.Server.Messaging.Messages;
using StealME.Server.Messaging.Requests;
using StealME.Server.Messaging.Responses;

namespace StealME.Server.Messaging
{
    public class TypeResolver : ITypeResolver
    {
        private static Dictionary<int, Type> _typeDictionary = new Dictionary<int, Type>() {
                                                                                               { 0,  typeof(TestRequest)    },
                                                                                               { 1,  typeof(TestResponse)   },

                                                                                               { 11, typeof(AuthRequest)    },
                                                                                               { 12, typeof(CommandRequest) },
                                                                                               { 13, typeof(GetRequest)     },
                                                                                               { 14, typeof(SetRequest)     },
                                                                                               { 15, typeof(PingRequest) },

                                                                                               { 21, typeof(ACKResponse)    },
                                                                                               { 22, typeof(AuthResponse)   },
                                                                                               { 23, typeof(StatusResponse) },
                                                                                               { 24, typeof(PingResponse) },
                                                                                               { 25, typeof(ProtocolVersionResponse) },

                                                                                               { 31, typeof(LocationMessage)},
                                                                                               { 32, typeof(TrackerStateMessage)},
                                                                                         };

        public Type[] GetTypes()
        {
            return _typeDictionary.Values.ToArray();
        }

        public Type GetTypeById(int id)
        {
            return _typeDictionary[id];
        }

        public int GetIdByType(Type type)
        {
            return _typeDictionary.First(a => a.Value == type).Key;
        }
    }
}

File: src/StealME.Client/src/stealme/client/networking/TypeResolver.java
package stealme.client.networking;

import java.util.Dictionary;
import java.util.HashMap;
import java.util.Map;
import java.util.Map.Entry;

public class TypeResolver {

	 Map<Integer, Class> _typeDictionary = new HashMap<Integer, Class>(){{
		    put(0, TestRequest.class);
		    put(1, TestResponse.class);
		    
		    put(11, AuthRequest.class);
		    put(12, CommandRequest.class);
		    put(13, GetRequest.class);
		    put(14, SetRequest.class);
		    put(15, PingRequest.class);
		    
		    put(21, ACKResponse.class);
		    put(22, AuthResponse.class);
		    put(23, StatusResponse.class);
		    put(24, PingResponse.class);
		    put(25, ProtocolVersionResponse.class);
		    
		    put(31, LocationMessage.class);
		    put(32, TrackerStateMessage.class);    
		}};
	 
   
	public Class[] GetTypes()
	{
		return (Class[]) _typeDictionary.values().toArray();
	}

	public Class GetTypeById(int id)
	{
		return _typeDictionary.get(id);
	}
	
	public int GetIdByType(Class type)
	{
		return getKeyByValue(_typeDictionary, type);
	}
	
	public static <T, E> T getKeyByValue(Map<T, E> map, E value) {
	    for (Entry<T, E> entry : map.entrySet()) {
	        if (value.equals(entry.getValue())) {
	            return entry.getKey();
	        }
	    }
	    return null;
	}
}
